measure.rb

```diff
@@ -1888,10 +1900,19 @@ def self.add_hot_water_and_appliances(runner, model, building, weather, spaces)
           capacity_kbtuh = 100000000.0
           oncycle_power = 0.0
           offcycle_power = 0.0
+
+          if has_desuperheater
+            if not related_hvac_list.include? relatedhvac
+              related_hvac_list << relatedhvac
+              coil_object = get_desuperheatercoil(@hvac_map, relatedhvac, sys_id)
+            else
+              fail "RelatedHVACSystem '#{relatedhvac}' for water heating system '#{sys_id}' is already attached to another water heating system."
+            end
+          end
           success = Waterheater.apply_tankless(model, runner, space, to_beopt_fuel(fuel),
                                                capacity_kbtuh, ef, cycling_derate,
                                                setpoint_temp, oncycle_power, offcycle_power, ec_adj,
-                                               @nbeds, @dhw_map, sys_id)
+                                               @nbeds, @dhw_map, sys_id, has_desuperheater, coil_object)
```

Comment:
In all the water heater methods, how about just passing the desuperheater coil_object?

Reply:
Do you mean adding a new "if not desuperheater_clg_coil.nil?" check in add_desuperheater function to avoid error for those systems without desuperheater?